app/controllers/admin/application_controller.rb

```diff
@@ -16,4 +17,20 @@ def authenticate_admin
   # def records_per_page
   #   params[:per_page] || 20
   # end
+
+  private
+
+  def no_access_guest
+    redirect_to new_user_session_path
+    no_access_flash
+  end
+
+  def no_access_user
+    redirect_to root_path
+    no_access_flash
```

Comment:
While this does work, it's more idiomatic to set the flash before doing the redirect.